packages/core/src/util/label.ts

```diff
@@ -25,9 +25,16 @@
 import startCase from 'lodash/startCase';
 
 import { ControlElement, LabelDescription } from '../models/uischema';
+import { JsonSchema } from '../models/jsonSchema';
 
-const deriveLabel = (controlElement: ControlElement): string => {
-  if (controlElement.scope !== undefined) {
+const deriveLabel = (
+  controlElement: ControlElement,
+  schemaElement?: JsonSchema
+): string => {
+  if (schemaElement && typeof schemaElement.title === 'string') {
+    return schemaElement.title;
+  }
+  if (typeof controlElement.scope === 'string') {
```

Comment:
why do we need this check?

Reply:
I did it because the check was already in the previous code as `controlElement.scope !== undefined`. I can remove it if you want.

Reply:
No, its fine. I missed the check that was there.